test/image/mocks/funnel_axis_with_other_traces.json

```diff
@@ -0,0 +1,196 @@
+            "orientation": "v",
+            "x": [
+                "A",
+                "B",
+                "C"
+            ],
+            "y": [
+                5,
+                4,
+                3
+            ],
+            "xaxis": "x",
+            "yaxis": "y"
+        },
+        {
+            "type": "scatter",
+            "orientation": "v",
+            "x": [
+                "A",
+                "B",
+                "C"
+            ],
+            "y": [
+                5,
+                4,
+                3
+            ],
+            "xaxis": "x",
+            "yaxis": "y"
+        },
+        {
+            "type": "funnel",
+            "width": 0.6,
+            "opacity": 0.8,
+            "orientation": "v",
+            "x": [
+                "A",
+                "B",
+                "C"
+            ],
+            "y": [
+                5,
+                4,
+                3
+            ],
+            "xaxis": "x2",
+            "yaxis": "y2"
+        },
+        {
+            "type": "bar",
+            "width": 0.4,
+            "orientation": "v",
+            "x": [
+                "A",
+                "B",
+                "C"
+            ],
+            "y": [
+                5,
+                4,
+                3
+            ],
+            "xaxis": "x2",
+            "yaxis": "y2"
+        },
+        {
+            "type": "funnel",
+            "width": 0.6,
+            "opacity": 0.8,
+            "orientation": "h",
+            "x": [
+                5,
+                4,
+                3
+            ],
+            "y": [
+                "A",
+                "B",
+                "C"
+            ],
+            "xaxis": "x3",
+            "yaxis": "y3"
+        },
+        {
+            "type": "waterfall",
+            "width": 0.4,
+            "orientation": "h",
+            "x": [
+                3,
+                -2,
+                0
+            ],
+            "y": [
+                "A",
+                "B",
+                "C"
+            ],
+            "measure": [
+                "r",
+                "r",
+                "t"
+            ],
+            "xaxis": "x3",
+            "yaxis": "y3"
+        },
+        {
+            "type": "funnel",
+            "width": 0.6,
+            "opacity": 0.8,
+            "orientation": "h",
+            "x": [
+                5,
+                4,
+                3
+            ],
+            "y": [
+                "A",
+                "B",
+                "C"
+            ],
+            "xaxis": "x4",
+            "yaxis": "y4"
+        }
+    ],
+    "layout": {
+        "width": 800,
+        "height": 800,
+        "dragmode": "pan",
+        "xaxis": {
+            "type": "category",
+            "domain": [
+                0,
+                0.48
+            ],
+            "autorange": true
+        },
+        "xaxis2": {
+            "type": "category",
```

Comment:
I _think_ you're missing `anchor` settings in this mock to make the y-axis labels not show up on top of one another.

We should have:

```
layout.xaxis2.anchor: 'y2'
layout.yaxis2.anchor: 'x2'
layout.xaxis3.anchor: 'y3'
layout.yaxis3.anchor: 'x3'
layout.xaxis4.anchor: 'y4'
layout.yaxis4.anchor: 'x3'
```

Reply:
Done in https://github.com/plotly/plotly.js/pull/3817/commits/c6155dbd1e2bd146848a0ada80cd7de44ada07ad.